Add Bootstrap Elements to Patient Visit Forms & Import Pages

Jdew192837 opened the pull request:
Fixes #876

Jdew192837 commented:
WIP for now.

Jdew192837 commented on interface/themes/style_light.css:
```diff
@@ -697,7 +697,7 @@ tr.odd, td.even {
   display: block;
 }
-.css_button[href*="deleter"], input[type="button"][href*="deleter"], button[href*="deleter"] {
+.css_button[href*="deleter"], input[type="button"][href*="deleter"], input[type="button"][class="deleter"], button[href*="deleter"] {
```
Allows for the red styling of 'Cancel' buttons. There are other areas in the codebase where this can be used.

Jdew192837 replied:
Will file an issue for this and fix it.

Jdew192837 commented on library/options.inc.php:
```diff
@@ -94,7 +94,7 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
         $tag_name_esc = $tag_name_esc . "[]";
     }
-    $s .= "<select name='$tag_name_esc'";
+    $s .= "<select name='$tag_name_esc' class='form-control' style='width: auto; display: inline-block'";
```
`form-control` is bootstrap
Everything else is for proper display, because bootstrap elements default to width: 100% and display: inline I believe

Jdew192837 commented on library/options.inc.php:
```diff
@@ -3232,7 +3232,7 @@ function genProviderSelect($selname, $toptext, $default=0, $disabled=false, $all
 
   $res = sqlStatement($query);
-  echo "   <select name='" . attr($selname) . "'";
+  echo "   <select class='form-control' style='width: auto; display: inline-block' name='" . attr($selname) . "'";
```
Same here

Jdew192837 commented:
Will upload screenshots once I am done with everything. If you all want to see them now, happy to screenshot for you

teryhill commented on interface/forms/misc_billing_options/new.php:
```diff
@@ -55,137 +56,223 @@
+    <?php call_required_libraries(['bootstrap']); ?>
+    <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>/jquery-min-1-9-1/index.js"></script>
+    <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>/bootstrap-3-3-4/dist/js/bootstrap.min.js"></script>
+    <script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery.datetimepicker.full.min.js"></script>
```
Jake you will need to use the "call_required_libraries" for these items also.
This example is from patient_tracker
'''<?php
    //  Include Bootstrap and DateTimePicker
  call_required_libraries(array("jquery-min-3-1-1","bootstrap","datepicker"));
?>
'''

teryhill replied:
Use the developer tools in chrome and you will see a console error

Jdew192837 replied:
Did I get it @teryhill ? My console looks fine now.

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -2,162 +2,203 @@
 <?php
 include_once("../../globals.php");
 include_once("$srcdir/api.inc");
+include_once("$srcdir/headers.inc.php");
```
Included bootstrap here

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -2,162 +2,203 @@
+	<head>
+		<?php html_header_show();?>
+		<link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
+		<!-- Get Bootstrap and jQuery (required for bootstrap) -->
```
Called Bootstrap & jQuery (required)

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -2,162 +2,203 @@
+		<link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
+		<!-- Get Bootstrap and jQuery (required for bootstrap) -->
+		<?php call_required_libraries(['bootstrap', 'jquery-min-1-9-1']); ?>
+		<script type="text/javascript">
```
Script to resize the panels to the same size

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -2,162 +2,203 @@
-<a href="<?php echo "$rootdir/patient_file/encounter/$returnurl";?>" class="link"
- onclick="top.restoreSession()">[<?php xl('Don\'t Save','e');?>]</a>
-</form>
+			<div class="col-sm-6 col-md-2 panel panel-primary">
```
Organized sections into panels, with clearer headers. Displays all panels in one tab. When multiple panels are open, it displays two columns next to each other to reduce congestion

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -2,162 +2,203 @@
+					<input type=checkbox name='foot_problems'  ><span class=text><?php xl('Foot Problems','e'); ?></span>
+				</div>
+			</div>
+			<h4><?php xl('Additional Notes: ','e'); ?></h4>
```
Increased size of the text

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -2,162 +2,203 @@
+				</div>
+			</div>
+			<h4><?php xl('Additional Notes: ','e'); ?></h4>
+			<textarea class="form-control" style="width: auto;" cols=40 rows=8 wrap=virtual name="additional_notes"></textarea><br>
```
Bootstrapped this

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -2,162 +2,203 @@
+			<div>
+				<!-- Save/Cancel buttons -->
+				<input type="button" id="save" value="<?php echo xla('Save'); ?>"> &nbsp;
+				<input type="button" id="dontsave" class="deleter" value="<?php echo xla('Cancel'); ?>"> &nbsp;
```
Made this red

Jdew192837 commented:
@teryhill @Trodrige PTAL at my latest commits. I think you'll like the way they look. :)

P.S. will screenshot them later, once everything is converted

aethelwulffe commented:
This is good looking work man!
Keep it up!  I would like to see a screenshot or two sooner than later.  I find that with UI stuff, it is a lot easier to get to the point where I feel like a test is in order when basic (obvious) things already look good...and sometimes I pull down a branch and forget what it is I am supposed to be looking for!
Advice:
-Keep white-space tight.  Ugly with everything on one page is easier to work with than pretty with scrolling when you are doing a lot of data entry.  There are plenty of people that do most of their work with <TAB>, but more that are "mouse and peck the keyboard".  Power users do most of the work, and are out priority.

Jdew192837 commented on interface/forms/ros/templates/ros/general_new.php:
```diff
@@ -1,1629 +1,1534 @@
-                                                         <?php } else {?>
-                                                            <label><input type="radio" name="jaundice" value="<?php echo $value;?>"/><?php echo $value;?></label>
-                                                         <?php }
-                                           
```
Resizes Panels to be height of the largest panel

Jdew192837 replied:
Made each section an individual panel

Jdew192837 replied:
The `get_peripheal` function is spelled wrong, but when changed to `get_peripheral` the function no longer works

Jdew192837 replied:
The `get_mearche` function is spelled wrong, but upon changing to `get_menarche` the function isn't responsive

Jdew192837 replied:
Bootstrap, Save Button

Jdew192837 replied:
Added `class="deleter'` to give the Cancel button a red color

Jdew192837 commented:
@teryhill @aethelwulffe Rewrote the ros/general_new.php
PTAL
Also could you assign this issue to me? Just so nothing happens :D

teryhill commented on interface/forms/ros/templates/ros/general_new.php:
```diff
@@ -1,1629 +1,1535 @@
-                                                         <?php } else {?>
-                                                            <label><input type="radio" name="jaundice" value="<?php echo $value;?>"/><?php echo $value;?></label>
-                                                         <?php }
-        
```
xlt respect gender as an improvement to the form.

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt . respect gender?

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
xlt

teryhill replied:
The function is in FormROS.class.php 
Change it there and here.

teryhill commented:
In general @Jdew192837  replace the xl 's with xlt 's you see what I mean

Jdew192837 commented:
Okay @terryhill thanks for the catch

On Dec 28, 2017 14:34, "Terry Hill" <notifications@github.com> wrote:

> In general @Jdew192837 <https://github.com/jdew192837> replace the xl 's
> with xlt 's you see what I mean
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LibreHealthIO/LibreEHR/pull/883#issuecomment-354354050>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXLk0crPKYfVbsTjHbUKH_7dKz8KrOk4ks5tE_tIgaJpZM4RLggV>
> .
>

aethelwulffe commented on interface/forms/ros/FormROS.class.php:
```diff
@@ -81,7 +81,7 @@ class FormROS extends ORDataObject {
     var $legpain_cramping = "N/A";
     var $history_murmur = "N/A";
-    var $arryhmia = "N/A";
+    var $arrythmia = "N/A";
```
arrhythmia...not arrythmia or arryhmia

Jdew192837 commented:
@aethelwulffe Should be good now. :)

Jdew192837 commented on interface/forms/reviewofs/new.php:
```diff
@@ -12,13 +12,15 @@
-		</script>
+	    <script type="text/javascript">
+	      // Resizes the panels to be the height of the largest panel
+	      function resizePanel() {
```
Separated function definition from call.

Jdew192837 commented on interface/forms/ros/FormROS.class.php:
```diff
@@ -81,7 +81,7 @@ class FormROS extends ORDataObject {
     var $legpain_cramping = "N/A";
     var $history_murmur = "N/A";
-    var $arrythmia = "N/A";
+    var $arryhthmia = "N/A";
```
Fixed spelling @aethelwulffe

Jdew192837 commented on interface/forms/ros/new.php:
```diff
@@ -1,6 +1,7 @@
 <?php
 include_once("../../globals.php");
 include_once("$srcdir/api.inc");
+include_once("$srcdir/headers.inc.php");
```
Included headers here to call bootstrap

aethelwulffe commented:
Then I will be testing shortly...after you post your screenshot of the finished product 😈

Jdew192837 commented:
@aethelwulffe I will only be able to apply after screenshots, so you can see the before :D

yehster commented on interface/forms/ros/templates/ros/general_new.php:
```diff
@@ -1528,8 +1526,8 @@
         <!-- Save/Cancel buttons -->
-        <input type="button" id="save" value="<?php echo xlta('Save'); ?>"> &nbsp;
-        <input type="button" id="dontsave" class="deleter" value="<?php echo xlta('Cancel'); ?>"> &nbsp;
+        <input type="button" id="save" value="<?php echo xlt('Save'); ?>"> &nbsp;
```
These translations should use xla as they generate HTML attributes.

Jdew192837 commented:
@yehster Where is this located?
Speech Dictation
What file is it?
Thanks,
jdew192837 (nulxis)

yehster commented on interface/forms/soap/templates/general_new.php:
```diff
@@ -68,8 +68,8 @@
 				<!-- Save/Cancel buttons -->
-				<input type="button" id="save" value="<?php echo xla('Save'); ?>"> &nbsp;
-				<input type="button" id="dontsave" class="deleter" value="<?php echo xla('Cancel'); ?>"> &nbsp;
+				<input type="button" id="save" value="<?php echo xlta('Save'); ?>"> &nbsp;
```
How did you test this change?

The function xlta does not exist.

xla is the correct

Jdew192837 commented:
@yehster I changed it already, take a look at the commits, I accidentally
added xlta, then corrected it
Sorry for the misunderstanding. And thanks for the help


On Fri, Dec 29, 2017 at 7:58 AM, Kevin Yeh <notifications@github.com> wrote:

> *@yehster* commented on this pull request.
> ------------------------------
>
> In interface/forms/soap/templates/general_new.php
> <https://github.com/LibreHealthIO/LibreEHR/pull/883#discussion_r159063750>
> :
>
> > @@ -68,8 +68,8 @@
>  			</table>
>  			<div>
>  				<!-- Save/Cancel buttons -->
> -				<input type="button" id="save" value="<?php echo xla('Save'); ?>"> &nbsp;
> -				<input type="button" id="dontsave" class="deleter" value="<?php echo xla('Cancel'); ?>"> &nbsp;
> +				<input type="button" id="save" value="<?php echo xlta('Save'); ?>"> &nbsp;
>
> How did you test this change?
>
> The function xlta does not exist.
>
> xla is the correct
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LibreHealthIO/LibreEHR/pull/883#pullrequestreview-85958305>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXLk0QsLyrfHa77YRaqJhVGJ_Vx42sZBks5tFPAYgaJpZM4RLggV>
> .
>

yehster commented on interface/forms/soap/templates/general_new.php:
```diff
@@ -68,8 +68,8 @@
-				<input type="button" id="save" value="<?php echo xlta('Save'); ?>"> &nbsp;
-				<input type="button" id="dontsave" class="deleter" value="<?php echo xlta('Cancel'); ?>"> &nbsp;
+				<input type="button" id="save" value="<?php echo xla('Save'); ?>"> &nbsp;
+				<input type="button" id="dontsave" class="deleter" value="<?php echo xla('Cancel'); ?>"> &nbsp;
```
Just for reference:
https://github.com/LibreHealthIO/LibreEHR/blob/master/library/htmlspecialchars.inc.php#L115

Jdew192837 commented:
@yehster Where is Speech Dictation located file name?

yehster commented:
https://github.com/LibreHealthIO/LibreEHR/tree/master/interface/forms/dictation

Jdew192837 commented:
Thanks @yehster

Jdew192837 commented:
I've got to head out for a little while, but all have been converted!! :) PTAL @teryhill @aethelwulffe @Trodrige

Jdew192837 commented:
Time for some screenshots:
===

Before (Misc Billing Options):
---
![before1](https://user-images.githubusercontent.com/24306897/34458637-cbdabb72-ed9f-11e7-906f-9e0c066515d6.png)

After (Misc Billing Options):
---
![after1](https://user-images.githubusercontent.com/24306897/34458639-d8d88138-ed9f-11e7-9e91-7327327d1efe.png)

Before (Procedure Order):
---
![before2](https://user-images.githubusercontent.com/24306897/34458641-e5b30dce-ed9f-11e7-91bd-4af1e59a97f3.png)

After (Procedure Order):
![after2](https://user-images.githubusercontent.com/24306897/34458642-ee3a9296-ed9f-11e7-9408-c551cb497fdf.png)

Before (Review of Systems):
---
![before3](https://user-images.githubusercontent.com/24306897/34458646-f966b190-ed9f-11e7-8dcf-d7c7d20e4e99.png)

After (Review of Systems):
---
![after3](https://user-images.githubusercontent.com/24306897/34458649-03b84b86-eda0-11e7-9ccf-efb5d9e0981f.png)

Before (Review of Systems Checks):
---
![before4](https://user-images.githubusercontent.com/24306897/34458650-0c3d2862-eda0-11e7-8f2a-74a86237c3c4.png)

After (Review of Systems Checks):
---
![after4](https://user-images.githubusercontent.com/24306897/34458651-19e62306-eda0-11e7-8ace-e513eeb4cbac.png)

Before (SOAP):
---
![before5](https://user-images.githubusercontent.com/24306897/34458652-26836fc4-eda0-11e7-923e-35a9efc4aa06.png)

After (SOAP):
---
![after5](https://user-images.githubusercontent.com/24306897/34458653-2e21249c-eda0-11e7-93f4-641823a58eed.png)

Before (Speech Dictation):
---
![before6](https://user-images.githubusercontent.com/24306897/34458654-392a0fa2-eda0-11e7-940c-948ec1f31ddf.png)

After (Speech Dictation):
---
![after6](https://user-images.githubusercontent.com/24306897/34458657-4d01ec48-eda0-11e7-9e3c-0e6dcd588426.png)

Before (Vitals):
---
![before7](https://user-images.githubusercontent.com/24306897/34458665-67f5d7c6-eda0-11e7-9c48-635cfa3a6479.png)

After (Vitals):
![after7](https://user-images.githubusercontent.com/24306897/34458667-6d4b4a76-eda0-11e7-9c7f-95c9178d780a.png)

Some of the Screenshots are missing what I edited, as I can't screenshot the whole page such as buttons, but you can take a look at yourselves. 
Sorry for the inconvenience.

Happy Holidays, and almost New Years
Thanks,
jdew192837

Jdew192837 commented:
@teryhill @robbyoconnor @aethelwulffe 
PTAL, provided you all with the coveted "Before" "After" screenshots!
Feel free to provide feedback!
Thanks,
jdew192837

teryhill commented:
@Jdew192837 Looks great only one small thing your after speech dictation is soap. I will look at the code after the holiday. @aethelwulffe  @tmccormi @Trodrige

Jdew192837 commented:
@teryhill @aethelwulffe PTAL at my final commits, and potentially merge. Thanks and happy new years!
jdew192837 (nulxis)

teryhill commented:
@aethelwulffe  can you look at this again.

Jdew192837 commented:
@aethelwulffe PTAL
Thanks,
jdew1982837

aethelwulffe commented:
Your after shot of speech dictation is actually the SOAP note....
No worry, all those forms should be replaced anyway :)
-Good to go here.....

Jdew192837 commented:
@aethelwulffe @tmccormi @teryhill 
PTAL, please so I may proceed witht he next issues, and get this task done :D.
Thanks,
jdew192837
